Refactor: Community/User avatars

hjiangsu opened the pull request:
## Pull Request Description

This PR performs a small refactor on the community and user avatars. Most notably:
- Renamed `CommunityIcon` to `CommunityAvatar` to match naming of the `UserAvatar`
- Moved `CommunityAvatar` and `UserAvatar` into `shared/avatars` directory
- Added more comments and documentation into the respective widgets
- Made `community` in `CommunityAvatar`, and `person` in UserAvatar non-nullable. Previously, the `community` and `person` parameters were nullable, but it was indicated as `required` in the constructor.

For example:
```dart
final Community? community;

const CommunityIcon({super.key, required this.community, this.radius = 12.0});
```

<!--- Please describe what was changed -->

## Issue Being Fixed

<!-- Please describe the problem that is being fixed and, if applicable, reference a GitHub issue -->

Issue Number: N/A

## Screenshots / Recordings

<!-- This section is optional but highly recommended to show off your changes! -->

## Checklist

- [ ] Did you update CHANGELOG.md?
- [ ] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

micahmo commented:
> * Made `community` in `CommunityAvatar`, and `person` in UserAvatar non-nullable. Previously, the `community` and `person` parameters were nullable, but it was indicated as `required` in the constructor.

I think we want to support a nullable community! For example, on the create post page before selecting a community, we want to have an "empty" avatar to fill the space. Otherwise it looks funny!

I think instead of making it non-nullable, it could be not required by the constructor.

| ![image](https://github.com/thunder-app/thunder/assets/7417301/90a174d7-9c28-4267-8e4d-8edca1d36c47) |
| - |

hjiangsu commented:
> For example, on the create post page before selecting a community, we want to have an "empty" avatar to fill the space. Otherwise it looks funny!

Good check! I reverted the change and removed `required` from the constructor

micahmo reviewed (approved):
LGTM!
